src/main/java/com/mineboundteam/minebound/inventory/AlloyFurnaceMenu.java

```diff
@@ -0,0 +1,225 @@
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.player.StackedContents;
+import net.minecraft.world.inventory.*;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.item.crafting.AbstractCookingRecipe;
+import net.minecraft.world.item.crafting.Recipe;
+import net.minecraft.world.item.crafting.RecipeType;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.AbstractFurnaceBlockEntity;
+
+public class AlloyFurnaceMenu extends RecipeBookMenu<Container> {  
+    public AlloyFurnaceMenu(int containerId, Inventory playerInventory, Container container, ContainerData data){
+        this(MenuRegistry.ALLOY_FURNACE_CONTAINER.get(), RecipeRegistry.ALLOY_FURNACE_RECIPE.get(), RecipeBookType.FURNACE, containerId, playerInventory, container, data);
+    }
+
+    public AlloyFurnaceMenu(MenuType<?> pMenuType, RecipeType<? extends AbstractCookingRecipe> pRecipeType, RecipeBookType pRecipeBookType, int pContainerId, Inventory pPlayerInventory, Container pContainer, ContainerData pData) {
+        super(pMenuType, pContainerId);
+        this.recipeType = pRecipeType;
+        this.recipeBookType = pRecipeBookType;
+        checkContainerSize(pContainer, 3);
+        checkContainerDataCount(pData, 4);
+        this.container = pContainer;
+        this.data = pData;
+        this.level = pPlayerInventory.player.level;
+
+        addSlots(pPlayerInventory);
+
+        addDataSlots(pData);
+    }
+
+    public AlloyFurnaceMenu(int containerId, Inventory playerInventory, FriendlyByteBuf buf) {
+        this(containerId, playerInventory, new SimpleContainer(5), new SimpleContainerData(4));
+    }
+
+    protected void addSlots(Inventory pPlayerInventory){
+        this.addSlot(new Slot(container, 0, 22, 34));
+        this.addSlot(new Slot(container, 3, 45, 21));
+        this.addSlot(new Slot(container, 4, 68, 34));
+        this.addSlot(new FuckingFuelSlot(this, container, 1, 45, 62));
+        this.addSlot(new FurnaceResultSlot(pPlayerInventory.player, container, 2, 139, 34));
+
+        addInventory(pPlayerInventory);
+        addHotBar(pPlayerInventory);
+    }
+
+    protected void addHotBar(Inventory playerInventory) {
+        for (int i = 0; i < 9; ++i) {
+            this.addSlot(new Slot(playerInventory, i, 8 + i * 18, 144));
+        }
+    }
+
+    protected void addInventory(Inventory playerInventory) {
+        for (int i = 0; i < 3; ++i) {
+            for (int j = 0; j < 9; ++j) {
+                this.addSlot(new Slot(playerInventory, j + i * 9 + 9, 8 + j * 18, 86 + i * 18));
+            }
+        }
+    }
+
+    @Override
+    public ItemStack quickMoveStack(Player pPlayer, int pIndex) {
+        ItemStack itemstack = ItemStack.EMPTY;
+        Slot slot = this.slots.get(pIndex);
+        if (slot != null && slot.hasItem()) {
```

Comment:
`slots.get()` will never be null so `slot != null` will always be true, can be removed